libsweep/src/unix/serial.cc

```diff
@@ -338,6 +338,8 @@ void device_read(device_s serial, void* to, int32_t len) {
         } else {
           throw error{"reading from serial device failed"};
         }
+      } else if(ret == 0){
+        throw error{"encountered EOF on serial device"};
```

Comment:
Sorry I'm a bit late to this, but can there be situations where we don't encounter EOF and a intermediate read returns 0? Or to be more precise: should we check `feof(serial->fd)` instead?